### PR TITLE
Postgres: Optional select clause elements and better `ON CONFLICT` support

### LINKED ISSUE #3365: Postgres: select targets are optional but SQLFluff does not support this
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

There is a special insert request that checks for the presence of a row before inserting, so as not to increase the primary key counter. However, sqlfluff cannot parse it.

### Expected Behaviour

Working without parsing errors.

### Observed Behaviour

```
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'SELECT'
L:   6 | P:   8 | L003 | Expected 1 indentations, found 0 [compared to line 05]
L:   8 | P:   2 |  PRS | Line 8, Position 2: Found unparsable section: '\nON
                       | CONFLICT DO NOTHING'
```

### How to reproduce

1) Create an `insert.sql` file:
```
INSERT INTO megatable (megacolumn)
SELECT * FROM (
    VALUES ( 'megavalue' )
) AS tmp (megacolumn)
WHERE NOT EXISTS (
SELECT FROM megatable AS mt
    WHERE mt.megacolumn = tmp.megacolumn
)
ON CONFLICT DO NOTHING;

```
2) Run `sqlfluff lint --dialect postgres insert.sql`

### Dialect

postgres

### Version

sqlfluff, version 0.13.1

### Configuration

[sqlfluff]
exclude_rules = L031

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L011]
aliasing = explicit

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #3365 

### Are there any other side effects of this change that we should be aware of?
I'm hoping not!

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 1]
```diff
@@ -1009,7 +1023,23 @@ class IntoClauseSegment(BaseSegment):
...
+        # here.
+        Ref("SelectClauseSegment"),
+        terminator=OneOf(
+            Ref("SetOperatorSegment"),
+            Ref("WithNoSchemaBindingClauseSegment"),
+            Ref("WithDataClauseSegment"),
+            Ref("OrderByClauseSegment"),
+            Ref("LimitClauseSegment"),
+            Ref("NamedWindowSegment"),
+            Sequence("ON", "CONFLICT"),
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_postgres.py:1039]: Could we (should we?) use copy/insert here?

File: test/fixtures/dialects/postgres/postgres_select.sql [hunk 2]
```diff
@@ -59,3 +59,5 @@ SELECT a < b COLLATE "de_DE" FROM test1;
 SELECT a COLLATE "de_DE" < b FROM test1;
 
 SELECT * FROM test1 ORDER BY a || b COLLATE "fr_FR";
+
+SELECT FROM test1;
```
Comment (barrywhart) [inline test/fixtures/dialects/postgres/postgres_select.sql:63]: What's this test for?

**Review by barrywhart — APPROVED** [review]

File: test/fixtures/dialects/postgres/postgres_select.sql (hunk 2 above)
Comment (tunetheweb) [reply test/fixtures/dialects/postgres/postgres_select.sql:63]: To test select clause elements are optional in Postgres (added in this change). Note it's `SELECT FROM` and not `SELECT * FROM`.

File: src/sqlfluff/dialects/dialect_postgres.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_postgres.py:1030]: Didn't think we could but figured it out. And cleaned some others up. Done!

**Pull request merged**